src/Sylius/Bundle/AdminBundle/Controller/DashboardController.php

```diff
@@ -38,72 +33,58 @@ final class DashboardController
 
     public function __construct(
-        DashboardStatisticsProviderInterface $statisticsProvider,
         ChannelRepositoryInterface $channelRepository,
         EngineInterface $templatingEngine,
         RouterInterface $router,
-        ?SalesDataProviderInterface $salesDataProvider = null
+        ?StatisticsController $statisticsController
     ) {
-        $this->statisticsProvider = $statisticsProvider;
         $this->channelRepository = $channelRepository;
         $this->templatingEngine = $templatingEngine;
         $this->router = $router;
-        $this->salesDataProvider = $salesDataProvider;
-
-        if ($this->salesDataProvider === null) {
-            @trigger_error(
-                sprintf('Not passing a $salesDataProvider to %s constructor is deprecated since Sylius 1.7 and will be removed in Sylius 2.0.', self::class),
-                \E_USER_DEPRECATED
-            );
-        }
+        $this->statisticsController = $statisticsController;
     }
 
     public function indexAction(Request $request): Response
     {
-        $channelCode = $request->query->get('channel');
-
         /** @var ChannelInterface|null $channel */
-        $channel = $this->findChannelByCodeOrFindFirst($channelCode);
+        $channel = $this->findChannelByCodeOrFindFirst($request->query->get('channel'));
 
         if (null === $channel) {
             return new RedirectResponse($this->router->generate('sylius_admin_channel_create'));
         }
-        // this data will be getting from UI after improve graph on dashboard
-        $startDate = (new \DateTime('first day of next month last year'));
-        $endDate = (new \DateTime('last day of this month'));
-
-        $statistics = $this->statisticsProvider->getStatisticsForChannelInPeriod($channel, $startDate, $endDate);
-        $data = ['statistics' => $statistics, 'channel' => $channel];
-
-        if ($this->salesDataProvider !== null) {
-            // this data will be getting from UI after improve graph on dashboard
-            $interval = Interval::month();
-
-            $data['sales_summary'] = $this
-                ->salesDataProvider
-                ->getSalesSummary($channel, $startDate, $endDate, $interval)
-            ;
-            $data['currency'] = $channel->getBaseCurrency()->getCode();
+
+        return $this->templatingEngine->renderResponse('@SyliusAdmin/Dashboard/index.html.twig', [
+            'channel' => $channel,
+        ]);
+    }
+
+    public function getRawData(Request $request): Response
+    {
+        /** @var ChannelInterface|null $channel */
+        $channel = $this->findChannelByCodeOrFindFirst($request->query->get('channel'));
+
+        if (null === $channel) {
+            return new RedirectResponse($this->router->generate('sylius_admin_channel_create'));
         }
 
-        return $this->templatingEngine->renderResponse('@SyliusAdmin/Dashboard/index.html.twig', $data);
+        return $this->statisticsController->getJsonData(
```

Comment:
Triggering of a controller in another controller is always a bad idea. The common logic needs to be extracted to separate service.